crates/anvil/src/eth/api.rs

```diff
@@ -1067,6 +1073,62 @@ impl EthApi {
         self.add_pending_transaction(pending_transaction, requires, provides)
     }
 
+    /// Waits for a transaction to be included in a block and returns its receipt.
+    /// Returns an error if the timeout is reached.
+    async fn check_inclusion(&self, hash: TxHash) -> Result<ReceiptResponse> {
```

Comment:
we can split this into two functions, one that simply does the loop and then

one more that wraps this with a given timeout, like await_inclusion_with_timeout(duration)